src/nodes/chunk_append/exec.c

```diff
@@ -293,9 +327,59 @@ chunk_append_begin(CustomScanState *node, EState *estate, int eflags)
 
 	initialize_constraints(state, lthird(cscan->custom_private));
 
+	/* In parallel mode with a parallel_aware plan, the parallel leader performs the startup
+	 * exclusion and stores the result in shared memory (the flag SUBPLAN_STATE_INCLUDED of
+	 * pstate->subplan_state is set for all included plans).
+	 *
+	 * The parallel workers use the information from shared memory to include the same plans as the
+	 * parallel leader. This ensures that all workers work on the same subplans and we have an
+	 * agreement about the number of subplans. This is necessary to ensure that the parallel workers
+	 * work correctly and that the next subplan to be processed in the shared memory
+	 * (pstate->next_plan) pointers to the same plan in all workers.
+	 *
+	 * If the workers perform the startup exclusion individually, they may choose different subplans
+	 * (e.g., due to a "constant" function that claims to be constant but returns different
+	 * results). In that case, we have a disagreement about the plans between the workers. This
+	 * would lead to hard-to-debug problems and out-of-bounds reads when pstate->next_plan is used
+	 * for subplan selection.
+	 *
+	 */
+	if (IsParallelWorker() && node->ss.ps.plan->parallel_aware)
+	{
+		/* The included subplans are initialized in chunk_append_initialize_worker. The needed
+		 * estate and eflags are stored to be able to perform the delayed initialization in
+		 * that function.
+		 *
+		 * Note: Due to force_parallel_mode, this node can be used in a ParallelWorker without
+		 * a parallel_aware plan. So, we have to check both conditions above to determine if
+		 * chunk_append_initialize_worker will be called or not and we can delay the
+		 * invokation of perform_plan_init().
```

Comment:
This comment sounds a little complicated to me. Maybe something like this:

We are inside a parallel worker running a parallel plan. Chunk exclusion was performed by the leader, and based on it, we will initialize the included subplans later, in chunk_append_initialize_worker. We have to store estate and eflags here that are needed for that initialization.

Note: When force_parallel_mode debug GUC is set, a normal sequential ChunkAppend plan can run inside a parallel worker. In this case, we have to perform the chunk exclusion right away. We distinguish it by that the `parallel_aware` flag of the plan is not set.

Reply:
@akuzm Thanks for the improved comment. It is changed in the PR.